lighthouse-core/lib/tracehouse/trace-processor.js

```diff
@@ -524,6 +524,28 @@ class TraceProcessor {
       firstMeaningfulPaint = lastCandidate;
     }
 
+    // LCP comes from the latest `largestContentfulPaint::Candidate`, but it can be invalidated
+    // by a `largestContentfulPaint::Invalidate` event. In the case that the last candidate is
+    // invalidated, the value will be undefined.
+    let largestContentfulPaint;
+    let invalidLcp = false;
```

Comment:
how about
```suggestion
    let lcpInvalidated = false;
```
to parallel `fmpFellBack` (and at least to me "invalidated" sounds like it has more specific criteria (like an `Invalidate` event) than just being invalid, which could conceivably include things like LCP events before navstart or a bad trace or something)